extensions/amp-subscriptions/0.1/local-subscription-platform-iframe.js

```diff
@@ -0,0 +1,174 @@
+ */
+export class LocalSubscriptionIframePlatform
+  extends LocalSubscriptionBasePlatform {
+
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   * @param {!JsonObject} platformConfig
+   * @param {!./service-adapter.ServiceAdapter} serviceAdapter
+   */
+  constructor(ampdoc, platformConfig, serviceAdapter) {
+    super(ampdoc, platformConfig, serviceAdapter);
+
+    devAssert(this.serviceConfig_['type'] == 'iframe',
+        'iframe initialized called without iframe config type');
+
+    /** @const @private {string} */
+    this.iframeSrc_ = userAssert(this.serviceConfig_['iframeSrc'],
+        '"iframeSrc" URL must be specified');
+    assertHttpsUrl(this.iframeSrc_, 'iframe Url');
+
+
+    /** @const @private {?Array} */
+    this.iframeVars_ = this.serviceConfig_['iframeVars'] || null;
+    if (this.iframeVars_) {
+      userAssert(isArray(this.iframeVars_),
+          '"iframeVars" must be an array');
+    }
+
+    /** @private @const {string} */
+    this.targetOrigin_ = parseUrlDeprecated(this.iframeSrc_).origin;
+
+    /** @private {?function()} */
+    this.connectedResolver_ = null;
+
+    /** @private {?Promise} */
+    this.connectedPromise_ = null;
+
+    /** @private @const {!Element} */
+    this.iframe_ = ampdoc.win.document.createElement('iframe');
+    toggle(this.iframe_, false);
+
+    /** @private @const {!Messenger} */
+    this.messenger_ = new Messenger(
+        this.ampdoc_.win,
+        () => this.iframe_.contentWindow,
+        this.targetOrigin_);
+
+    /** @private {?Promise<!JsonObject>} */
+    this.configPromise_ = null;
+
+    this.initializeListeners_();
+  }
+
+  /** @override */
+  getEntitlements() {
+    return this.connect().then(() => {
+      return this.messenger_.sendCommandRsvp('authorize', {})
+          .then(res => JSON.stringify(res))
+          .then(resJson => Entitlement.parseFromJson(resJson));
+    });
+  }
+
+  /** @override */
+  isPingbackEnabled() {
+    return true;
+  }
+
+  /** @override */
+  pingback(selectedEntitlement) {
+    return this.connect().then(() => {
+      return this.messenger_.sendCommandRsvp(
+          'pingback', {entitlement: selectedEntitlement});
+    });
+  }
+
+  /**
+   * @return {!Promise}
+   * @package Visible for testing only.
+   */
+  connect() {
+    if (!this.connectedPromise_) {
+      const deferred = new Deferred();
+      this.connectedPromise_ = deferred.promise;
+      this.connectedResolver_ = deferred.resolve;
+
+      this.configPromise_ = this.resolveConfig_();
+      // Connect.
+      this.messenger_.connect(this.handleCommand_.bind(this));
+      this.ampdoc_.getBody().appendChild(this.iframe_);
+      this.iframe_.src = this.iframeSrc_;
+    }
+    return this.connectedPromise_;
+  }
+
+  /**
+   * @return {!Promise<!JsonObject>}
+   * @private
+   */
+  resolveConfig_() {
+    return new Promise(resolve => {
+      const configJson = parseJson(JSON.stringify(this.serviceConfig_));
```

Comment:
I'm wondering if we also should include the actual `pageConfig` from LD-Schema/microdata here. Or at least the key fields such as `productID`, `publicationID`?

Reply:
added to config:

```
pageConfig: {
          encryptedDocumentKey: <String|null>,
          productId: '...',
          publicationId: '...',
        },
```